### PR TITLE
[Bounty] getGameByID()

### PR DESCRIPTION
**Pull request by CoeJoder**
## Summary
Added `getGameByID()` function & 14 data-driven integration tests.
## Other Information
Let me know if you need additional changes.

### CODE REVIEW

**Review by andyruwruw — APPROVED** [review]
Looks good to me!

@pi0neerpat @jschiarizzi if everything looks good to the two of you, I'll merge and publish!

File: src/endpoints/games.js [hunk 1]
```diff
@@ -98,27 +98,81 @@ function _getPGN(pgnHeaders, moveList) {
...
-    let encTo = moveList[i + 1];
-    chess.move({
-      'from': _decodeMove(encFrom),
-      'to': _decodeMove(encTo)
-    });
+    let move = {
+      'from': _decodeMove(moveList[i], false),
+      'to': _decodeMove(moveList[i + 1], true)
+    }
+    if (move.to instanceof _PawnPromotion) {
```
Comment (jschiarizzi) [inline src/endpoints/games.js:105]: Detects pawn promotion looks good

File: src/endpoints/games.js [hunk 2]
```diff
@@ -98,27 +98,81 @@ function _getPGN(pgnHeaders, moveList) {
...
+          fromFileIndex + 1 :
+          fromFileIndex;
+
+  // sanity check: ensure pawn is still on the board after promotion
+  if (toFileIndex < 0 || toFileIndex > BOARD_LENGTH - 1) {
+    throw new Error(`Invalid pawn promotion; file index out of bounds: ${toFileIndex}`);
+  }
+  const toFile = BOARD_FILES[toFileIndex];
+
+  // sanity check: ensure pawn rank is 2 or 7 prior to promotion
```
Comment (jschiarizzi) [inline src/endpoints/games.js:163]: Good sanity check here. Possible also future on that makes sure new piece is on rank 1 or 8

File: tests/games.test.js [hunk 3]
```diff
@@ -1,36 +1,166 @@
...
+// long game
 // API: https://www.chess.com/callback/live/game/6508402266
 // WEB: https://www.chess.com/analysis/game/live/6508402266
 const TEST_GAME_2 = [
   // id, startFen, endFen
   '6508402266',
   'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1',
-  'K1r5/p5kp/8/6PP/8/1r6/8/8 w - - 9 56'
-]
+  'K1r5/p5kp/8/6PP/8/1r6/8/8 w - - 9 56',
```
Comment (jschiarizzi) [inline tests/games.test.js:23]: Thank you for writing more tests, fabulous

**Review by jschiarizzi — APPROVED** [review]

**Comment by andyruwruw** [comment]
Just noticed the bounty was paid! Merging the branch. Thank you @CoeJoder!

**Pull request merged**